cloud-sql/sql-server/sqlalchemy/main.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import logging
+import os
+
+from flask import Flask, render_template, request, Response
+import sqlalchemy
+from sqlalchemy import Table
+from sqlalchemy import Column
+from sqlalchemy import DateTime, Integer, String
+
+app = Flask(__name__)
+
+logger = logging.getLogger()
+
+# [START cloud_sql_server_python_sqlalchemy_create]
+# Remember - storing secrets in plaintext is potentially unsafe. Consider using
+# something like https://cloud.google.com/kms/ to help keep secrets secret.
+db_user = os.environ.get("DB_USER")
+db_pass = os.environ.get("DB_PASS")
+db_name = os.environ.get("DB_NAME")
+
+# If this is running in the docker container when deployed to GAE flex,
+# use "172.17.0.1"
+host = "172.17.0.1" if os.environ.get("PROD") else "127.0.0.1"
+
+# The SQLAlchemy engine will help manage interactions, including automatically
+# managing a pool of connections to your database
+db = sqlalchemy.create_engine(
+    # Equivalent URL:
+    # mssql+pyodbc://<db_user>:<db_pass>@/<host>:<port>/<db_name>?driver=ODBC+Driver+17+for+SQL+Server
+    sqlalchemy.engine.url.URL(
+        "mssql+pyodbc",
+        username=db_user,
+        password=db_pass,
+        database=db_name,
+        host=host,
+        port=1433,
+        query={"driver": "ODBC Driver 17 for SQL Server"},
```

Comment:
Does this driver support unix socket?

Reply:
nope :/ 
https://docs.microsoft.com/en-us/sql/connect/odbc/linux-mac/connection-string-keywords-and-data-source-names-dsns?view=sql-server-ver15